### ISSUE TITLE
(feature request) Support for conversion of std::filesystem::path

### ISSUE THREAD

**Issue by shartte** [muellan/clipp#7]
Since CLIs often have to deal with files and directories being passed to them on the command line, the C++17 filesystem classes make life a lot easier.
It would be nice to be able to just do this:

```c++
#include <filesystem>
std::filesystem::path input_file;
auto cli = (
		value("input file", input_file),
);
```

As far as I can tell, this would require another specialization of the make<> template.
While the issue is not really all that pressing for the official library (vendors are still working on officially implementing the C++17 library extensions, although std::experimental::filesystem does exist for VS2017), is there a documented way of providing user-supplied specializations of make<>?

**Comment by muellan** [comment]
I'm also looking forward to the day when we have ```std::filesystem``` in all major standard library implementations. As far as I understand the [interface](http://en.cppreference.com/w/cpp/filesystem/path/path) of ```std::filesystem::path```, it has a constructor, that can convert character sequences like ```const char*``` or ```std::string``` to a path.
So what I would need to do (and I don't know why I didn't already) is to add a *default* for ```clipp::detail::make```:
```cpp
namespace clipp {  namespace detail {
//...
template<class T>
struct make {
    static inline T from(const char* s) {
        if(!s) return false;
        return static_cast<T>(s);
    }
};
//...
}  }
```
Apart from that, you can always write your own convenience factory functions:
```cpp
#include <filesystem>

template<class Target>
clipp::parameter //or auto
filename(const std::string& label, std::filesystem::path& path) {
    return clipp::value(label).call([&](const char* arg) {  //or std::string
        path = arg; //should work AFAIK
    });
}

std::filesystem::path infile;
auto cli = (
    filename("input file", infile)
);
```

**Comment by muellan** [comment]
I just added a default conversion for the ```make<>``` template, that should work with ```std::filesystem::path```.

**Comment by muellan** [comment]
I think I could also add ```filename```, ```opt_filename```, ```filenames``` and ```opt_filenames``` as convenience value-parameter factories in the future that will only take refs to ```std::filesystem::path``` or ```std:vector<std::filesystem::path>``` and will be available ```#if __cplusplus >= 201703L```.

**Comment by shartte** [comment]
Thanks alot for the detailed answers! I went with a custom factory for the time being.
I am not sure about __cplusplus since I am unsure whether it also applies to library conformance, or just the language bits.

**Comment by muellan** [comment]
Yeah, you also have to check for ```#if__has_include(<filesystem>)``` and probably something else ... have to check that.

**Comment by shartte** [comment]
I guess I wouldn't put too much thought into it heh. Even though I am using the lib in my project now, the custom factory approach really is a good enough "workaround".
I also looked into this a bit and found several feature detection recommendations, among them testing for `__cpp_lib_filesystem`, but I haven't seen how that would be used correctly.
One of the issues is that libstdc++ seems to have <filesystem>, but still hides the namespaces behind std::experimental.
